.github/workflows/tests.yml

```
@@ -0,0 +1,50 @@
+name: test
+
+on:
+  push:
+    branches: [master, unstuck-vmprof]
+  pull_request:
+    branches: [master, unstuck-vmprof]
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    permissions:
+      pull-requests: write
+    name: ${{ matrix.os }} - ${{ matrix.python }}
+    strategy:
+      fail-fast: true
```

Maybe run all the CI for now?
```suggestion
      fail-fast: false
```